packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -92,17 +135,58 @@ const TokenListRow = ({ closeAction, token }: TokenListRowProps) => (
   </div>
 );
 
-const TokenList = ({ closeAction, tokenList }: TokenListProps) => (
-  <div
-    className="mt-5 overflow-y-auto h-72 border-t border-surface-50 divide-y divide-surface-50"
-    id="tokenPickerList"
-  >
-    {tokenList.map(token => (
-      <TokenListRow
-        closeAction={closeAction}
-        key={token.address}
-        token={token}
-      />
-    ))}
-  </div>
-);
+const TokenList = ({
+  closeAction,
+  onClearSearchClick,
+  tokenList,
+  tokenSearchQuery
+}: TokenListProps) => {
+  const [filteredTokenList, setFilteredTokenList] = useState<Token[]>(
+    tokenList
+  );
+
+  const handleClearSearchClick = () => {
+    onClearSearchClick();
+    setFilteredTokenList(tokenList);
+  };
+
+  useEffect(() => {
+    if (tokenSearchQuery) {
+      const filteredItems = tokenList.filter(
+        token =>
+          token.symbol.toLowerCase().includes(tokenSearchQuery.toLowerCase()) ||
+          token.name.toLowerCase().includes(tokenSearchQuery.toLowerCase())
```

Comment:
I'm okay with this code but will leave an idea if you find it interesting:


```js
const checkMatch = (matchString, searchQuery) => {
 matchString.toLowerCase().includes(searchQuery.toLowerCase())
}
```

```
checkMatch(token.symbol) || checkMatch(token.name) 
```

something like this would remove some repetition.

Reply:
Fine with both. But I would think in a better naming for the function like `matches` or `stringIncludes`.  Also, no function no naming troubles x)

Reply:
yeah, the names can improve! and yeah, agree, bad naming might be better no function!

Reply:
If you don't mind it, I rather leave it as it is right now and abstract it in the future if needed. 

Although I'm a big fan of abstracting repeated code, in this particular case we will end up with more lines of code (because `tokenSearchQuery` may be falsy, thus typechecking workarounds)